sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py

```diff
@@ -781,7 +781,7 @@ def create_or_update(
         :param entity: The resource to create or update.
         :type entity: typing.Union[~azure.ai.ml.entities.Job
             , ~azure.ai.ml.entities.Model, ~azure.ai.ml.entities.Environment, ~azure.ai.ml.entities.Component
-            , ~azure.ai.ml.entities.Datastore, ~azure.ai.ml.entities.WorkspaceAssetReference]
```

Comment:
This also needs to be removed from TypeVar "T" on line 772

Reply:
Thanks for pointing this out, now removed.